File: matriz.js
import promptSync from "prompt-sync";
const prompt = promptSync();

let alumnos = [];


function agregarAlumno() {
    let nombre = prompt("Ingrese el nombre del alumno: ");

    // Verificar si el alumno ya existe
    for (let i = 0; i < alumnos.length; i++) {
        if (alumnos[i][0].toLowerCase() === nombre.toLowerCase()) {
            console.log("El alumno ya existe.");
            return;
        }
    }

    let materias = ["Matematicas", "Lengua", "Sociales", "Naturales"];
    let notas = [];

    // Pedir notas para cada materia
    for (let i = 0; i < materias.length; i++) {
        let nota;
        while (true) {
            nota = parseInt(prompt("Ingrese la nota de " + materias[i] + " (0 a 10): "));
            if (!isNaN(nota) && nota >= 0 && nota <= 10) break;
            console.log("Por favor ingrese una nota válida entre 0 y 10.");
        }
        notas.push([materias[i], nota]);
    }

    alumnos.push([nombre, notas]);
    console.log("Alumno agregado con éxito.");
}

// Pedir cantidad de alumnos
let cantidad = parseInt(prompt("Ingrese la cantidad de alumnos a cargar: "));
console.log(" ");

for (let i = 0; i < cantidad; i++) {
    console.log("Carga los datos del alumno numero " + (i + 1) + " de " + cantidad + "");
    agregarAlumno();
}

// Mostrar todos los alumnos al final
console.log("Listado de alumnos cargados: ");
console.log(" ")
for (let i = 0; i < alumnos.length; i++) {
    let nombre = alumnos[i][0];
    let notas = alumnos[i][1];
    console.log("Alumno: " + nombre);
    
    for (let j = 0; j < notas.length; j++) { // Mostrar cada materia y su nota
        console.log("  " + notas[j][0] + ": " + notas[j][1]);
    }
}

File: matriz.py
alumnos = []

contador = int(input("Ingrese la cantidad de alumnos a cargar: "))

def agregar_alumno():
    nombre = input("Ingrese el nombre del alumno: ")

    # Verificar si el alumno ya existe
    for alumno in alumnos:
        if alumno[0].lower() == nombre.lower():
            print("El alumno ya existe.")
            return

    materias = ["Matematicas", "Lengua", "Sociales", "Naturales"]
    notas = []

    # Bucle para cargar las notas de cada materia
    for materia in materias:
        while True:
            try:
                nota = int(input(f"Ingrese la nota de {materia}: "))
                if 0 <= nota <= 10:
                    break
                else:
                    print("La nota debe estar entre 0 y 10.")
            except ValueError:
                print("Por favor ingrese un número válido.")
        notas.append([materia, nota])

    alumnos.append([nombre, notas])
    print("Alumno agregado con éxito.\n")

# Bucle principal para cargar todos los alumnos
for i in range(contador):
    print(f" Carga los datos del alumno numero {i + 1} de {contador} ")
    agregar_alumno()

# Mostrar todos los alumnos cargados
print("lista de alumnos cargados: ")
print(" ")
for alumno in alumnos:
    nombre = alumno[0]
    notas = alumno[1]
    print(f"Alumno: {nombre}")
    for materia, nota in notas:
        print(f" {materia}: {nota}")
